### DISCUSSION TITLE
How to pass RBAC file in docker-compose.yml

### DISCUSSION THREAD

**Discussion by g-fusion** [concourse/concourse#6618] [Help & Support]
Hello,

I know it is probably straight forward to do that, but how should I pass file for [rbac config](https://concourse-ci.org/user-roles.html#configuring-rbac) to [docker-compose](https://github.com/concourse/concourse-docker/blob/master/docker-compose.yml#L23-L30) as env? I can see env property from  [web spec](https://github.com/concourse/concourse-bosh-release/blob/6cc9cf39d7172bf4858c0984f02e42479a12b874/jobs/web/templates/bpm.yml.erb#L287), but passing just CONCOURSE_CONFIG_RBAC: /path/to/rbac/config.yml does not work. I could not find any examples of this for docker, howerver such exists for [helm chart](https://github.com/concourse/concourse-chart/blob/master/values.yaml#L927-L935). What do I miss?
Thanks.

**Comment by taylorsilva** [comment]
One way with docker would be to write the file to disk and pass it in as a volume. I assume you're doing this locally on your machine

1. Make your config.yml with your desired contents
2. In the same or parent directory should be your docker-compose.yml file. In that file, add the directory that your config.yml is in as a volume. Here's a quick example:
```
mkdir rbac
vim rbac/config.yml
```
Then in your docker-compose add a line to the web's volumes like:
```yaml
    volumes:
    - ./rbac:/rbac
```
And then in the `environment` section you can add
```
CONCOURSE_CONFIG_RBAC: /rbac/config.yml
```

**Comment by g-fusion** [comment]
@taylorsilva - It works. 
However, maybe a little hint in documentation will be good to avoid such questions in the future. I also noticed that [quick start ](https://concourse-ci.org/quick-start.html) guide for Docker does not contain the part of key generation mentioned [here](https://github.com/concourse/concourse-docker#running-with-docker-compose).
Thanks.

**Reply by taylorsilva** [reply]
Totally agree. There are lots of quality-of-life improvements that could be made to the docs.
